File: src/item.cpp
#include <QVector>

#include "item.h"

namespace orq
{
	Item::Item()
	{
    }

	bool Item::saveChanges()
	{
		bool success = false;
		return success;
	}

	bool Item::compare(Item item)
	{
		bool success = false;
		return success;
	}

    QVector<Item> Item: getChildren()
	{
        auto vector = QVector<Item>();
        return vector;
	}

	Item Item ::getParent()
	{
        auto item = Item();
		return item;
	}

	bool Item::setParent(Item item)
	{
		bool success = false;
		return success;
	}

	bool Item:: edit()
	{
        bool success = false;
        return success;
    }
}

File: openrq/item.py

from PySide2.QtWidgets import QString, QVector, QByteArray

class Item():
	def __init__(self):
		self.id = 0
		self.uid = 0
        self.version = 0
        self.shown = True
        self.description = QString()
        self.name = ""
        self.contains_color = [Red, Green, Blue, Purple]
        
	def saveChanges(self):
		pass

	def getChildren(self):
		pass
	
	def addChild(self, child):
		pass
	
	def removeChild(self, child):
		pass

	def getHash(self, QByteArray):
		pass